packages/react-virtualizer/src/components/Virtualizer/Virtualizer.component-browser-spec.tsx

```diff
@@ -0,0 +1,348 @@
+    const items = await component.locator('[role="listitem"]').count();
+    expect(items).toBe(0);
+
+    // Add first item
+    await component.getByTestId('add-item-button').click();
+    await page.waitForTimeout(500);
+
+    // Should have one item now
+    const firstItem = component.getByTestId('item-0');
+    await expect(firstItem).toBeVisible({ timeout: 3000 });
+    await expect(firstItem).toHaveAttribute('data-value', '0');
+    await expect(firstItem).toContainText('Item 0');
+
+    // Add second item (should be prepended)
+    await component.getByTestId('add-item-button').click();
+    await page.waitForTimeout(500);
+
+    // item-1 should now be first (prepended)
+    const newFirstItem = component.getByTestId('item-1');
+    await expect(newFirstItem).toBeVisible({ timeout: 3000 });
+    await expect(newFirstItem).toHaveAttribute('data-value', '1');
+
+    // item-0 should still be visible but below item-1
+    await expect(firstItem).toBeVisible();
+
+    // Verify no React key reconciliation errors
+    const keyErrors = consoleErrors.filter(
+      (err) =>
+        err.includes('key') ||
+        err.includes('unique') ||
+        err.includes('Warning: Encountered two children with the same key')
+    );
+
+    expect(keyErrors).toHaveLength(0);
+  });
+
+  test('should handle prepending multiple items without React key errors', async ({
+    mount,
+    page,
+  }) => {
+    const consoleErrors: string[] = [];
+    page.on('console', (msg) => {
+      if (msg.type() === 'error') {
+        consoleErrors.push(msg.text());
+      }
+    });
+
+    const component = await mount(<VirtualizerExample numItems={0} />);
+
+    await expect(component.getByTestId('scroll-container')).toBeVisible();
+    await page.waitForTimeout(500);
+
+    // Add 10 items at once
+    await component.getByTestId('add-multiple-button').click();
+    await page.waitForTimeout(500);
+
+    // Should have items visible
+    const itemsAfterFirst = await component
+      .locator('[role="listitem"]')
+      .count();
+    expect(itemsAfterFirst).toBeGreaterThan(0);
+
+    // First visible item should be item-0 (the first of the 10 added)
+    const firstItem = component.getByTestId('item-0');
+    await expect(firstItem).toBeVisible({ timeout: 3000 });
+    await expect(firstItem).toHaveAttribute('data-value', '0');
+
+    // Add 10 more items (should be prepended)
+    await component.getByTestId('add-multiple-button').click();
+    await page.waitForTimeout(500);
+
+    // Now item-10 should be first
+    const newFirstItem = component.getByTestId('item-10');
+    await expect(newFirstItem).toBeVisible({ timeout: 3000 });
+    await expect(newFirstItem).toHaveAttribute('data-value', '10');
+
+    // No key reconciliation errors
+    const keyErrors = consoleErrors.filter(
+      (err) =>
+        err.includes('key') ||
+        err.includes('unique') ||
+        err.includes('Warning: Encountered two children with the same key')
+    );
```

Comment:
even if it's a test, you can make it more DRY and descriptive 

make a function outside of the tests, something as this: 

```ts
const hasKeyErrors = (errors: string[]) => {
    const regex = /(key|unique|Encountered two children with the same key)/i;
    return errors.some(err => regex.test(err));
};

// test
expect(hasKeyErrors(consoleErrors)).toBe(false);
```